src/home_robot_hw/home_robot_hw/nodes/goto_controller.py

```diff
@@ -119,12 +120,15 @@ def _run_control_loop(self):
         rate = rospy.Rate(self.hz)
 
         while not rospy.is_shutdown():
-            if self.active and self.xyt_goal is not None:
-                # Compute control
-                v_cmd, w_cmd = self.controller.compute_control()
-
-                # Command robot
-                self._set_velocity(v_cmd, w_cmd)
+            if self.xyt_goal is not None:
+                if self.active:
+                    # Compute control
+                    v_cmd, w_cmd = self.controller.compute_control()
+
+                    # Command robot
+                    self._set_velocity(v_cmd, w_cmd)
+                else:
+                    log.warn("Received a goal while NOT active. Goal will be unset.")
```

Comment:
why not unset the goal here? self.xyt_goal = None

Reply:
Austin suggested that unsetting goal in _enable_service ensures controller always starts in neutral state, which makes sense. That said, I think we should unset goal here as well so it doesn't keep triggering this condition. I'll add this tomorrow.